src/gangliad/statsd.cpp

```diff
@@ -987,21 +1057,138 @@ StatsD::publishMetrics( int /* timerID */ )
 		}
 	}
 
-	CollectorList* collectors = daemonCore->getCollectorList();
-	ASSERT( collectors );
+	// Now that we have formulated the query, go out and query collector(s)
+	
+	// Handle config knob MONITOR_MULTIPLE_COLLECTORS, which is an optional knob to
+	// tell the gangliad to monitor a list of specified collectors instead of the
+	// default pool collector.  This knob should be a comma seperated list of entries,
+	// where each entry is name-of-collector/address-of-collector, where the address is
+	// same as what you would give the "-pool" parameter of condor_status.  For instance,
+	// MONITOR_MULTIPLE_COLLECTORS = ce1.wisc.edu/ce1.wisc.edu:9619, ce2.wisc.edu/ce2.wisc.edu:9619
+	bool multiple_collectors = false;
+	std::vector<std::string> collector_pool_list;
+	std::vector<std::string> collector_name_list;
+	char *param_monitor_multiple_collectors = NULL;
+	if (m_param_monitor_multiple_collectors.empty()) {
+		param_monitor_multiple_collectors = param("MONITOR_MULTIPLE_COLLECTORS");
+	} else {
+		param_monitor_multiple_collectors = strdup(m_param_monitor_multiple_collectors.c_str());
+	}
+	if (param_monitor_multiple_collectors && param_monitor_multiple_collectors[0]) {
+		StringList collist(param_monitor_multiple_collectors);
+		free(param_monitor_multiple_collectors);
+		param_monitor_multiple_collectors = NULL;
+		collist.rewind();
+		char *col;
+		while( (col=collist.next()) ) {
+			char* pos = strchr(col,'/');
+			if (!pos || !pos[1]) {
+				break;
+			}
```

Comment:
Good catch re the malformed list inconsistency.  Instead of changing the `break` to a `continue`, I will push a commit that just does an `EXCEPT`.  My thinking here is if an entry in MONITOR_MULTIPLE_COLLECTORS is malformed (the expectation is every entry will have a '/'), throw an error about it -vs- just continue and ignore.